contribute/code.rst

```diff
@@ -0,0 +1,213 @@
+Code changes
+============
+
+This page helps you get your changes to Ubuntu Touch's code merged. To help you with this, we describe the things we look for in code, commits, and changesets. We also take a look at what to expect in the code review process.
+
+Assumptions
+-----------
+
+This page does not include information about how to develop applications or changes to Ubuntu Touch. To learn how to develop Ubuntu Touch components, see the :doc:`/systemdev/index` section. To learn how to develop applications, see the :doc:`/appdev/index` or :doc:`/contribute/preinstalled-apps` sections.
+
+This page assumes you know how to write your changes, use Git to commit them, push them to GitHub or Gitlab as needed, fork a repository, and make a Pull Request (PR) on GitHub or Merge Request (MR) on GitLab. We are only covering what you need to put in your changes, commit messages, and changeset descriptions; and how to expect the review process will work.
+
+Definitions
+-----------
+
+Since we use GitHub and GitLab for our code hosting depending on the component, some terms must be defined up front for both platforms.
+
+changeset
+    A GitHub Pull Request (PR) or GitLab Merge Request (MR).
+issue report
+    A GitHub or GitLab Issue.
+
+Reasoning
+---------
+
+Ubuntu Touch is used by many users on many different devices. Breaking functionality that these users rely on can be catastrophic. Our changeset review process helps us avoid lasting broken functionality by ensuring a few properties of the change.
+
+In particular, we ask: Does the change...
+
+* meet its own goals, such as adding a feature or fixing a bug?
+* cause any unexpected behavior?
+* make it easy for someone to make further changes to the software in the future?
+
+Sometimes we make mistakes. We may accept a change which causes unexpected behavior or does not meet its own goals. In that case, the other properties can help us decide what to do next. We might revert the changeset, going back to the state before the change was made. We might also build on top of the old changeset. The decision can be made quickly and without breaking more functionality as long as our review process was followed.
+
+The rules on this page explain how you should write and document your changes to Ubuntu Touch. Following these rules makes the review process easier for us. An easier review means your changes are accepted and deployed to the many users of Ubuntu Touch more quickly.
+
+These are not absolute; we won't automatically reject your change based on whether you follow the rules. Make sure you can explain your decision to ignore one or more rules when asked, though.
```

Comment:
I find it easier to understand when turned around: 

"Make sure .... though" -> "If you break a rule, make sure you can explain why."